scala-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/scala/ScalaInterpreter.scala

```diff
@@ -228,16 +228,15 @@ class ScalaInterpreter() extends Interpreter {
     urls.foldLeft("")((l, r) => ClassPath.join(l, r.toString))
   }
 
-  protected def interpreterArgs(kernel: KernelLike): List[String] = {
+  protected def interpreterArgs(): List[String] = {
     import scala.collection.JavaConverters._
-    kernel.config.getStringList("interpreter_args").asScala.toList
-  }
-
-  protected def maxInterpreterThreads(kernel: KernelLike): Int = {
-    kernel.config.getInt("max_interpreter_threads")
+    if(config.hasPath("interpreter_args"))
+      config.getStringList("interpreter_args").asScala.toList
```

Comment:
nope... it fails. And there is no API in `Config` to get Options or pass defaults. An approach I could take to clean this up is to do
```
config.withFallback(ConfigFactory.parse(Map("interpreter_args"->""))).getStringList("interpreter_args")
```